src/main/java/io/jenkins/plugins/opentelemetry/job/MonitoringRunListener.java

```diff
@@ -273,7 +274,15 @@ public void _onFinalized(Run run) {
                     parentSpan.setAttribute(JenkinsOtelSemanticAttributes.CI_PIPELINE_RUN_DESCRIPTION, description);
                 }
                 parentSpan.setAttribute(JenkinsOtelSemanticAttributes.CI_PIPELINE_RUN_RESULT, runResult.toString());
-                StatusCode statusCode = Result.SUCCESS.equals(runResult) ? StatusCode.OK : StatusCode.ERROR;
+
+                StatusCode statusCode;
+                if (Result.SUCCESS.equals(runResult)) {
+                    statusCode = StatusCode.OK;
+                } else {
+                    parentSpan.setAttribute(SemanticAttributes.EXCEPTION_TYPE, "PIPELINE_" + runResult);
+                    parentSpan.setAttribute(SemanticAttributes.EXCEPTION_MESSAGE, "PIPELINE_" + runResult);
+                    statusCode = StatusCode.ERROR;
```

Comment:
I had to deal with something similar for the ansible integration (https://github.com/ansible-collections/community.general/pull/3837/files), so I used UNSET.
I don't know if this could be potentially someething similar. For some reason I think it can help to identify verified from unverified failures. While unverified failures are related to flakiness, environmental issues. So nobuilt and aborted won't be treated as unverified failures but event.outcome=unknown/unset

So maybe this distinction can help to categorise builds in three sets, rather than two?

Reply:
Do you mean setting statusCode to `UNSET` when the Jenkins job result is not_built or aborted?

Do you mean this logic (pseudo code):
```
switch (run.result) {
case null:
case NOT_BUILT:
case ABORTED:
   span.statuscode = UNSET
   break;
case SUCCESS:
   span.statuscode = OK
   break
case UNSTABLE:
case FAILURE:
   span.statuscode = ERROR 

}